x/programs/rust/examples/nft/src/lib.rs

```diff
@@ -93,55 +95,52 @@ pub fn init(
 /// Mints NFT tokens and sends them to the recipient.
 #[public]
 pub fn mint(program: Program, recipient: Address) -> bool {
-    const MINT_AMOUNT: i64 = 1;
-
     let counter = program
         .state()
         .get::<i64, _>(StateKey::Counter.to_vec())
-        .expect("failed to store ");
+        .unwrap_or_default();
 
-    assert!(counter <= TOTAL_SUPPLY, "max supply for nft exceeded");
+    assert!(counter <= MAX_SUPPLY, "max supply for nft exceeded");
 
     let balance = program
         .state()
-        .get::<i64, _>(StateKey::Balance(recipient).to_vec())
+        .get::<i64, _>(StateKey::Balances(recipient).to_vec())
         .unwrap_or_default();
 
     program
         .state()
         .store(
-            StateKey::Balance(recipient).to_vec(),
+            StateKey::Balances(recipient).to_vec(),
             &(balance + MINT_AMOUNT),
         )
         .expect("failed to store balance");
 
     program
         .state()
-        .store(StateKey::Owner.to_vec(), &recipient)
+        .store(StateKey::Owners(recipient).to_vec(), &recipient)
         .expect("failed to store owner");
 
     program
         .state()
-        .store(StateKey::Counter.to_vec(), &(counter + 1_i64))
+        .store(StateKey::Counter.to_vec(), &(counter + MINT_AMOUNT))
         .is_ok()
 }
 
 #[public]
 pub fn burn(program: Program, from: Address) -> bool {
-    const BURN_AMOUNT: i64 = 1;
     let null_address = Address::new([0; 32]);
 
     // Only the owner of the NFT can burn it
     let owner = program
```

Comment:
I don't understand what this check is for? Won't this just fail if the person burning doesn't have a balance?